Lib/pickle.py

```diff
@@ -1604,17 +1604,26 @@ def load_dup(self):
 
     def load_get(self):
         i = int(self.readline()[:-1])
-        self.append(self.memo[i])
+        try:
+            self.append(self.memo[i])
+        except KeyError as exc:
+            raise UnpicklingError from exc
```

Comment:
Can you set the exception context to be "from None" to be the same as the C implementation?